src/UI/class-settings-page.php

```diff
@@ -930,44 +941,42 @@ public function validate_options( $input ) {
 	 * @return ParselySettingOptions Validated inputs.
 	 */
 	private function validate_basic_section( $input ) {
-		$options = $this->parsely->get_options();
+		$are_credentials_managed = Parsely::are_credentials_managed();
+		$options                 = $this->parsely->get_options();
 
-		if ( '' === $input['apikey'] ) {
-			add_settings_error(
-				Parsely::OPTIONS_KEY,
-				'apikey',
-				__( 'Please specify the Site ID', 'wp-parsely' )
-			);
+		if ( $are_credentials_managed ) {
+			$input['apikey']     = '';
+			$input['api_secret'] = '';
 		} else {
-			$site_id = $this->sanitize_site_id( $input['apikey'] );
-			if ( false === $this->validate_site_id( $site_id ) ) {
+			if ( '' === $input['apikey'] ) {
 				add_settings_error(
 					Parsely::OPTIONS_KEY,
 					'apikey',
-					__( 'Your Parse.ly Site ID looks incorrect, it should look like "example.com".', 'wp-parsely' )
+					__( 'Please specify the Site ID', 'wp-parsely' )
 				);
 			} else {
-				$input['apikey'] = $site_id;
+				$site_id = $this->sanitize_site_id( $input['apikey'] );
+				if ( false === $this->validate_site_id( $site_id ) ) {
+					add_settings_error(
+						Parsely::OPTIONS_KEY,
+						'apikey',
+						__( 'The Site ID was not saved because it is incorrect. It should look like "example.com".', 'wp-parsely' )
+					);
+					$input['apikey'] = $options['apikey'];
+				} else {
+					$input['apikey'] = $site_id;
+				}
 			}
-		}
-
-		$input['api_secret'] = $this->get_unobfuscated_value( $input['api_secret'], $this->parsely->get_api_secret() );
 
-		$input['metadata_secret'] = $this->get_unobfuscated_value( $input['metadata_secret'], $this->parsely->get_options()['metadata_secret'] );
-		if ( '' !== $input['metadata_secret'] ) {
-			if ( strlen( $input['metadata_secret'] ) !== 10 ) {
+			$input['api_secret'] = $this->get_unobfuscated_value( $input['api_secret'], $this->parsely->get_api_secret() );
+			$api_secret_length   = strlen( $input['api_secret'] );
+			if ( $api_secret_length > 0 && $api_secret_length <= 30 ) {
```

Comment:
This was briefly discussed as a simple way to prevent mixing up the api and metadata secret. It's not a comprehensive check but would address some portion of tickets.

Reply:
Yes, this has been mentioned several times in different occasions. It is indeed quite simplistic, but it will prevent a customer from injecting shorter keys within the API Secret field (thing which has happened lately according to Parse.ly support).

Ideally in the future we can have real validation with an API call or such.